### ISSUE TITLE
Missing support Ubuntu 18.04 LTS (Bionic Beaver)

### ISSUE THREAD

**Issue by ad-m** [hwdsl2/setup-ipsec-vpn#377]
There is missing support for Ubuntu 18.04 LTS (Bionic Beaver). I think it is desirable to support this version of the operating system.

After upgrade from Ubuntu 16.04 I got following error:
```
$ sudo systemctl status ipsec.service
● ipsec.service - Internet Key Exchange (IKE) Protocol Daemon for IPsec
   Loaded: loaded (/lib/systemd/system/ipsec.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Tue 2018-05-01 12:19:35 CEST; 4min 48s ago
     Docs: man:ipsec(8)
           man:pluto(8)
           man:ipsec.conf(5)
 Main PID: 6783 (code=exited, status=127)

May 01 12:19:35 tacjana systemd[1]: ipsec.service: Service hold-off time over, scheduling restart.
May 01 12:19:35 tacjana systemd[1]: ipsec.service: Scheduled restart job, restart counter is at 5.
May 01 12:19:35 tacjana systemd[1]: Stopped Internet Key Exchange (IKE) Protocol Daemon for IPsec.
May 01 12:19:35 tacjana systemd[1]: ipsec.service: Start request repeated too quickly.
May 01 12:19:35 tacjana systemd[1]: ipsec.service: Failed with result 'exit-code'.
May 01 12:19:35 tacjana systemd[1]: Failed to start Internet Key Exchange (IKE) Protocol Daemon for IPsec.
```
I am debugging why.

**Comment by ad-m** [comment]
```
$ /usr/local/libexec/ipsec/pluto --leak-detective --config /etc/ipsec.conf --nofork 
/usr/local/libexec/ipsec/pluto: error while loading shared libraries: libevent-2.0.so.5: cannot open shared object file: No such file or directory
```

**Comment by ad-m** [comment]
There is need to rebuild IPSec. I made them manually analyzing the code:
```
SWAN_VER=3.23
swan_file="libreswan-$SWAN_VER.tar.gz"
swan_url1="https://github.com/libreswan/libreswan/archive/v$SWAN_VER.tar.gz"
swan_url2="https://download.libreswan.org/$swan_file"
if ! { wget -t 3 -T 30 -nv -O "$swan_file" "$swan_url1" || wget -t 3 -T 30 -nv -O "$swan_file" "$swan_url2"; }; then
  exiterr "Cannot download Libreswan source."
fi
/bin/rm -rf "/opt/src/libreswan-$SWAN_VER"
tar xzf "$swan_file" && /bin/rm -f "$swan_file"
cd "libreswan-$SWAN_VER" || exiterr "Cannot enter Libreswan source dir."
sed -i '/docker-targets\.mk/d' Makefile
cat > Makefile.inc.local <<'EOF'
WERROR_CFLAGS =
USE_DNSSEC = false
EOF
if [ "$(packaging/utils/lswan_detect.sh init)" = "systemd" ]; then
  apt-get -yq install libsystemd-dev || exiterr2
fi
NPROCS="$(grep -c ^processor /proc/cpuinfo)"
[ -z "$NPROCS" ] && NPROCS=1
make "-j$((NPROCS+1))" -s base && make -s install-base
```

Next to reload:
```
sudo systemctl start ipsec.service
```

Now everything looks good:
```
$ sudo systemctl status ipsec.service
● ipsec.service - Internet Key Exchange (IKE) Protocol Daemon for IPsec
   Loaded: loaded (/lib/systemd/system/ipsec.service; enabled; vendor preset: enabled)
   Active: active (running) since Tue 2018-05-01 12:43:24 CEST; 3s ago
     Docs: man:ipsec(8)
           man:pluto(8)
           man:ipsec.conf(5)
  Process: 16237 ExecStartPre=/usr/local/sbin/ipsec --checknflog (code=exited, status=0/SUCCESS)
  Process: 16236 ExecStartPre=/usr/local/sbin/ipsec --checknss (code=exited, status=0/SUCCESS)
  Process: 16027 ExecStartPre=/usr/local/libexec/ipsec/_stackmanager start (code=exited, status=0/SUCCESS)
  Process: 16026 ExecStartPre=/usr/local/libexec/ipsec/addconn --config /etc/ipsec.conf --checkconfig (code=exited, st
 Main PID: 16248 (pluto)
   Status: "Startup completed."
    Tasks: 4 (limit: 4728)
   CGroup: /system.slice/ipsec.service
           └─16248 /usr/local/libexec/ipsec/pluto --leak-detective --config /etc/ipsec.conf --nofork

May 01 12:43:25 tacjana pluto[16248]: adding interface lo/lo 127.0.0.1:4500
May 01 12:43:25 tacjana pluto[16248]: adding interface lo/lo ::1:500
May 01 12:43:25 tacjana pluto[16248]: | setup callback for interface lo:500 fd 22
May 01 12:43:25 tacjana pluto[16248]: | setup callback for interface lo:4500 fd 21
May 01 12:43:25 tacjana pluto[16248]: | setup callback for interface lo:500 fd 20
May 01 12:43:25 tacjana pluto[16248]: | setup callback for interface eth0:4500 fd 19
May 01 12:43:25 tacjana pluto[16248]: | setup callback for interface eth0:500 fd 18
May 01 12:43:25 tacjana pluto[16248]: | setup callback for interface tun0:4500 fd 17
May 01 12:43:25 tacjana pluto[16248]: | setup callback for interface tun0:500 fd 16
May 01 12:43:25 tacjana pluto[16248]: loading secrets from "/etc/ipsec.secrets"
```

**Comment by hwdsl2** [comment]
@ad-m Hello! Glad to hear that it worked. This issue seems to be specific to your system which upgraded from 16.04. As an alternative to re-building Libreswan, you may also try re-creating the symbolic link. See [1].

[1] http://www.nigeldunn.com/2011/12/11/libevent-2-0-so-5-cannot-open-shared-object-file-no-such-file-or-directory/

**Comment by friek** [comment]
@hwdsl2 this has been fixed in xl2tpd 1.3.12.
